crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1555,8 +1568,34 @@ impl<'db> KnownClass {
             Self::GenericAlias | Self::ModuleType | Self::FunctionType => {
                 types_symbol(db, self.as_str()).unwrap_or_unknown()
             }
-            Self::SpecialForm => typing_symbol(db, self.as_str()).unwrap_or_unknown(),
             Self::NoneType => typeshed_symbol(db, self.as_str()).unwrap_or_unknown(),
+            Self::SpecialForm => typing_symbol(db, self.as_str()).unwrap_or_unknown(),
+            Self::TypeVar => typing_symbol(db, self.as_str()).unwrap_or_unknown(),
+            Self::NoDefaultType => typing_extensions_symbol(db, self.as_str()).unwrap_or_unknown(),
+        }
+    }
+
+    fn is_singleton(self) -> bool {
```

Comment:
```suggestion
    const fn is_singleton(self) -> bool {
```

Reply:
Oops, already had hit the auto-merge button. I'll push a follow-up for this.